cds-snc/covid-alert-portal
# Generate QR Code poster

# Summary | Résumé

Generates the QR Code Poster. For now, the poster is not converted to PDF or sent by email. This PR uses an SVG template file to generate a visual representation of the poster, and renders it at a URL that is logged to the console. A future PR will convert this output to PDF and attach it to an email

When this is deployed, a new environment variable will need to be added with the Private Signing Key, and the Public Key will need to be shared with the Mobile App team for payload verification.

## Not included
- Doesn't convert to PDF
- Doesn't send the email
- Doesn't include terraform changes for environment variable (yet)
- This is not the final poster design, notably it does not render/print at the correct dimensions

## To test
- Be sure to follow the instructions in the next section to Create a signing key
- Go through the process of validating your email, then fill out the location details
- Once you hit "Make poster" on the Summary screen, you will see a URL output on the console. It will look like this: `http://localhost:8000/en/locations/7cb5422c-2c43-45f1-9882-1857cc283e3e/poster`
- Click the link to see the poster

## Create a signing key
You will need to set an environment variable `QRCODE_SIGNATURE_PRIVATE_KEY` which contains a private key for signing the QR Code payload. You can generate one for testing purposes.

From the command line enter a python shell. Import the utils file from the register app, and run generate_signature_key():

```
> python
Python 3.9.1 (default, Feb  3 2021, 07:38:02)
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from register import utils
>>> utils.generate_signature_key()
'g0xsyM4EJQqBD5zSPE6a4ORVq+wIpr3dWNx75q0otQQ='
```

Use the key output for the environment variable. Warning that this will not work with the Mobile App at this time as we will need to share the public key with the App team once we generate production keys.
